configure.ac

```diff
@@ -802,6 +802,23 @@ dnl ========================================================================
 AC_CHECK_FUNCS(getopt, AC_DEFINE(HAVE_DECL_GETOPT,  1, [Have getopt function]))
 AC_CHECK_FUNCS(nanosleep, AC_DEFINE(HAVE_DECL_NANOSLEEP,  1, [Have nanosleep function]))
 
+AC_CACHE_CHECK(whether sscanf supports %m,
+               pf_cv_var_sscanf,
+               AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
+#include <stdio.h>
+const char *s = "some-command-line-arg";
+int main(int argc, char **argv) {
+char *name = NULL;
+sscanf(s, "%b", &name);
```

Comment:
Presumably you meant %m?

Reply:
Yes, that should not have gotten committed.  I was using that to check to verify the test fails.  It does not, which makes me feel less good about how this is all going to work on older OSes.  You can make it fail with -Wformat -Werror, but I don't think that would be supported on older OSes either.

Reply:
You should be able to check the return value, it'll be 1 on successful match and something else otherwise.

Reply:
Doing so requires using the AC_RUN_IFELSE macro instead of the AC_COMPILE_IFELSE macro.  The autoconf docs warn that means cross-compiling will not work unless you provide a default value.  It also suggests you provide a pessimistic default.  In this case, I'd go with defaulting to not supporting %m.

Reply:
Makes sense

Reply:
That would convenient, but it would miss too many cases. Many features that eventually made it into POSIX 2008 (including %m) were originally implemented much earlier.

Digression: I haven't found a tidy matrix of what POSIX standard/features were supported by what versions of different compilers/standard libraries. That would make it much easier to decide when we can require certain POSIX version compatibility upstream. Another complication is that claiming capability with a certain POSIX standard does not guarantee that every aspect of that standard has been implemented (as a trivial example, see errno(3) for EOPNOTSUPP on Linux).

Reply:
Yeah, I think the suggestion is basically:
```
if _POSIX_C_SOURCE >= 200809L:
    success
else if running compiled program works:
    success
else:
    failure
```

That is, the posix test only allows for the test to succeed.  It alone does not allow for the test to fail.  This would also prevent this from being a problem should we ever care about cross compilation.

Reply:
Something like this:
```
AC_CACHE_CHECK(whether sscanf supports %m,
               pf_cv_var_sscanf,
               AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
#include <unistd.h>
int main(int argc, char **argv) {
#if defined _POSIX_C_SOURCE && _POSIX_C_SOURCE >= 200809L
return 0;
#else
#error POSIX support is too old
return 1;
#endif
}
]])],
                                 pv_cv_var_sscanf="yes",
                                 AC_RUN_IFELSE([AC_LANG_SOURCE([[
#include <stdio.h>
const char *s = "some-command-line-arg";
int main(int argc, char **argv) {
char *name = NULL;
int n = sscanf(s, "%ms", &name);
return n == 0 ? 1 : 0;
}
]])],
                                                pf_cv_var_sscanf="yes", pf_cv_var_sscanf="no", pf_cv_var_sscanf="no")))

if test "$pf_cv_var_sscanf" = "yes"; then
    AC_DEFINE(SSCANF_HAS_M, 1, [ ])
fi
```

Reply:
That would be fine, but is there any advantage over just the run test?

Reply:
It avoids running a compiled program as part of the compilation process on the majority of machines, which could be nice.  We don't especially care about cross-compilation, but if the posix test succeeds then there wouldn't be an impediment to cross-compilation here.  I don't really care one way or another, though.